### ISSUE TITLE
Questions regarding the experiment on civilcomments

### ISSUE THREAD

**Issue by yangarbiter** [facebookresearch/BalancingGroups#4]
I have some questions regarding the group assignments for the `civilcomments` dataset. In [the paper](https://arxiv.org/pdf/2110.14503.pdf), it is mentioned that coarse grouping is being used, so there should only be two groups (the example being one of the identities or not).

I have generated the metadata (`metadata_civilcomments_coarse.csv`) with [setup_datasets.py](https://github.com/facebookresearch/BalancingGroups/blob/main/setup_datasets.py). In this metadata file, it appears that there are 8 different groups in the column named "a" (value 0-7), which seems to be inconsistent with the two groups mentioned in the paper. From [the code](https://github.com/facebookresearch/BalancingGroups/blob/main/setup_datasets.py#L155-L162), it appears that only the training set is using the coarse grouping while the validation and testing sets are using fine grouping. I am wondering why it is designed this way or whether there are any places that I have misunderstood?

Thank you for your time in advance.

**Comment by BadrYoubiIdrissi** [comment]
The columns a in meta `metadata_civilcomments_coarse.csv` does indeed contain the 0-7 values but only for the "val" and "test" split. For the "train" split there should only be two values 0 and 1 as you pointed out in the code you linked. The paper mentions the coarse grained dataset only for training purposes. This was mainly done to have the same experimental conditions as the JTT papers and be able to reproduce their results. The idea of using these groups only for training is to accomodate gDRO which cannot be trained with the finer groups that overlap. On the other algorithms that allow the use of the fine grained attributes that overlap i.e. SUBG and RWG we do use the fine grained dataset ([code](https://github.com/facebookresearch/BalancingGroups/blob/c9a15f32b8e18cf00c15d9d0c4853789b49e0df8/datasets.py#L320-L322)). Since the original goal is to perform well on all the groups in the fine grained dataset we evaluate all the algorithms on that. If we evaluated gDRO on the coarse grained dataset, its worst group accuracy wouldn't be comparable to the other algorithms'. 

Hope this clarifies some things ! Sorry for the delayed answer :)

Here some code I ran to check
![image](https://user-images.githubusercontent.com/18900192/149488686-8230b1cb-2027-41c3-a360-37d8e968762d.png)
